mlflow/openai/__init__.py

```diff
@@ -844,3 +862,31 @@ def load_model(model_uri, dst_path=None):
     _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
     model_data_path = os.path.join(local_model_path, flavor_conf.get("data", MODEL_FILENAME))
     return _load_model(model_data_path)
+
+
+@experimental
+@autologging_integration(FLAVOR_NAME)
+def autolog(
+    log_input_examples=False,
+    log_model_signatures=False,
+    log_models=False,
+    log_datasets=False,
+    disable=False,
+    exclusive=False,
+    disable_for_unsupported_versions=True,
+    silent=False,
+    registered_model_name=None,
+    extra_tags=None,
+):
+    with contextlib.suppress(ImportError):
+        from openai.resources.chat.completions import Completions as ChatCompletions
+        from openai.resources.completions import Completions
+        from openai.resources.embeddings import Embeddings
+
+        for task in (ChatCompletions, Completions, Embeddings):
+            safe_patch(
+                FLAVOR_NAME,
+                task,
+                "create",
+                patched_call,
+            )
```

Comment:
Is it challenging to patch methods in `openai.OpenAI`?

Reply:
Not really challenging. `openai.chat.completions` and `openai.OpenAI().chat.completions` are both objects of the same class, so we just need to patch the `create()` function of that class.

Reply:
Got it! Can we do that?

Reply:
or the current implementation already does that?

Reply:
the current implementation already does that. I cleaned up some of the code

Reply:
currently it patches `openai.resources.chat.completions.Completions`, which is the class of both `openai.chat.completions` and `openai.OpenAI().chat.completions`